File: GymMembershipClient/Models/Member.cs
using System.Collections.Generic;

namespace GymMembershipClient.Models
{
    public class Member
    {
        public long Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public Subscription Subscription { get; set; }
        public List<ClassEnrollment> Enrollments { get; set; }
    }
}

File: backend/src/main/java/com/gym_membership/entity/Member.java
package com.gym_membership.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "members")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Member extends User {
    
    @OneToOne(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    @lombok.ToString.Exclude
    @lombok.EqualsAndHashCode.Exclude
    private Subscription subscription;

    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    @lombok.ToString.Exclude
    @lombok.EqualsAndHashCode.Exclude
    private List<ClassEnrollment> enrollments = new ArrayList<>();

    private String firstName;
    private String lastName;
    private String phoneNumber;
    private String profilePicture;
}
